src/Dropbox/DropboxFile.php

```diff
@@ -8,193 +8,241 @@
-            // Read from the offset till the maxLength
-            return $this->stream->read($this->maxLength);
-        }
-
-        return $this->stream->getContents();
-    }
-
-    /**
-     * Get the name of the file
-     *
-     * @return string
-     */
-    public function getFileName()
-    {
-        return basename($this->path);
-    }
-
-    /**
-     * Get the path of the file
-     *
-     * @return string
-     */
-    public function getFilePath()
-    {
-        return $this->path;
-    }
-
-    /**
-     * Get the size of the file
-     *
-     * @return int
-     */
-    public function getSize()
-    {
-        return $this->stream->getSize();
-    }
-
-    /**
-     * Get mimetype of the file
-     *
-     * @return string
-     */
-    public function getMimetype()
-    {
-        return \GuzzleHttp\Psr7\mimetype_from_filename($this->path) ?: 'text/plain';
-    }
-
-    /**
-     * Returns true if the path to the file is remote
-     *
-     * @param string $pathToFile
-     *
-     * @return boolean
-     */
-    protected function isRemoteFile($pathToFile)
-    {
-        return preg_match('/^(https?|ftp):\/\/.*/', $pathToFile) === 1;
-    }
+	/**
+	 * Path of the file to upload
+	 *
+	 * @var string
+	 */
+	protected $path;
+
+	/**
+	 * The maximum bytes to read. Defaults to -1 (read all the remaining buffer).
+	 *
+	 * @var int
+	 */
+	protected $maxLength;
+
+	/**
+	 * Seek to the specified offset before reading.
+	 * If this number is negative, no seeking will
+	 * occur and reading will start from the current.
+	 *
+	 * @var int
+	 */
+	protected $offset;
+
+	/**
+	 * File Stream
+	 *
+	 * @var \GuzzleHttp\Psr7\Stream
+	 */
+	protected $stream;
+
+	/**
+	 * @param     $filePath
+	 * @param int $maxLength
+	 * @param int $offset
+	 *
+	 * @return \Kunnu\Dropbox\DropboxFile
+	 */
+	public static function createByPath($filePath, $maxLength = -1, $offset = -1)
+	{
+		$obj = new self($maxLength, $offset);
+		$obj->setPath($filePath);
+		$obj->open();
+		return $obj;
+	}
+
+	/**
+	 * @param     $filename
+	 * @param     $resource
+	 * @param int $maxLength
+	 * @param int $offset
+	 *
+	 * @return \Kunnu\Dropbox\DropboxFile
+	 * @throws \Kunnu\Dropbox\Exceptions\DropboxClientException
+	 */
+	public static function createByStream($filename, $resource, $maxLength = -1, $offset = -1)
+	{
+		$obj    = new self($maxLength, $offset);
+		$obj->setPath($filename);
+		$stream = \GuzzleHttp\Psr7\stream_for($resource);
+		if(!$stream) {
+			throw new DropboxClientException('Failed to create DropboxFile instance. Unable to open resource.');
+		}
+		$obj->setStream($stream);
+		return $obj;
+	}
+
+	/**
+	 * Create a new DropboxFile instance
+	 *
+	 * @param int $maxLength Max Bytes to read from the file
+	 * @param int $offset    Seek to specified offset before reading
+	 *
+	 * @throws \Kunnu\Dropbox\Exceptions\DropboxClientException
+	 */
+	private function __construct($maxLength = -1, $offset = -1)
```

Comment:
This introduces a breaking change to the public API of the DropboxFile class. Instead of removing the first parameter (that takes in the `filepath`) and changing the access to private, you can check if the `filepath` passed is a `resource` (then you can probably call the `createByStream` method) or a `string` (call the `createByPath` method).